beacon_node/network/src/network_beacon_processor/gossip_methods.rs

```diff
@@ -1019,9 +994,7 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
             .process_gossip_data_columns(vec![verified_data_column])
             .await
         {
-            Ok((availability, data_columns_to_publish)) => {
-                self.handle_data_columns_to_publish(data_columns_to_publish);
-
+            Ok(availability) => {
```

Comment:
You could add a match here to re-assign result
```rust
let result = match result {
  MissingComponents => self.attempt_data_column_reconstruction(),
  other => other,
};
```
to re-use the handling of successful conditions. Same could apply to sync

Reply:
Not quite sure what you mean, as we don't use the `availability` result after this, are you meaning to comment on sync? or do you want to extract the handling of successful condition to a method?

Reply:
Right, for `process_gossip_verified_data_column` it's not that useful and the function is simple. The way you mutate `result` now should be fine